src/gui/editors/SongEditor.cpp

```diff
@@ -318,7 +318,14 @@ void SongEditor::keyPressEvent( QKeyEvent * ke )
 						ke->key() == Qt::Key_Insert )
 	{
 		m_song->insertBar();
-	}
+    #ifdef Q_OS_MACOS
+        if( /*_ke->modifiers() & Qt::ShiftModifier*/
+           gui->mainWindow()->isShiftPressed() == true &&
+                            ke->key() == Qt::Key_Enter || ke->key() == Qt::Key_Return )
+        {
+            m_song->insertBar();
+        }
+    #endif
```

Comment:
1. I think you accidentally removed a curly bracket here.
2. We use [LMMS_BUILD_APPLE](https://github.com/LMMS/lmms/blob/stable-1.2/src/lmmsconfig.h.in#L4) as our precompiler directive for MacOS.
3. You should group the `||` logic with parens to be explicit.
4. @LMMS/developers are there any objections to just removing the precompiler and doing `( ke->key() == Qt::Key_Insert || Qt::Key_Enter || ke->key() == Qt::Key_Return)`  for all platforms?

Reply:
Yeah, I noticed that I accidentally removed the curly bracket - was planning on just creating a new PR.
We should probably test Return on macOS platforms before universally changing it, tho - if it doesn't work, Shift+D and Shift+I would be good options to fulfill the same purpose (as discussed previously).

Reply:
> there any objections to just removing the precompiler and doing[…] for all platforms?

No objections, sounds like a good idea. It's simpler and avoids confusion for people who switch platforms.

Reply:
No objections, too.

Reply:
No objections.

Reply:
Should we proceed with integrating this change into the master branch, then?

Reply:
Yes. At this point this will be going into master. @tresf

Reply:
A thought just crossed my mind... has anyone actually *tested* these particular key combos on macOS? I'm currently running a build that @tresf graciously compiled for me, but it uses Shift+Command to insert bars instead of the designated key combos suggested here.

Reply:
I've since started using a Mac keyboard and the lack of delete shortcut is a huge hindrance.  Any small shortcut changes that fix an OS-specific bug are fine in stable.  It's unfortunate that the OP has abandoned the PR so if the best we can do is land it in master, oh well.

Reply:
haha, I'm still here - also, Delete is Fn+Shift+Delete (pretty sure that we talked about this in the Discord chat); this PR just needs testing, which I can probably later today/tomorrow (wasn't able to setup my dev environment prior to macOS Mojave's GA)